cgen error when compiling with `-d use_libbacktrace`

### Describe the bug

Running `./v -g -o vdbg cmd/v && ./vdbg -d use_libbacktrace 'foo.v' && 'foo'` on `foo.v` produced a compiler error.

### Reproduction Steps

```sh
./v -g -o vdbg cmd/v && ./vdbg -d use_libbacktrace 'foo.v' && 'foo'
```

```v
fn main() {
	panic('aaaa')
}
```

### Expected Behavior

No C error

### Current Behavior

```
> Failed build_thirdparty_obj_file cmd
>           mod: builtin
>          path: /home/ge/.vlang/thirdparty/libbacktrace/backtrace.o
>   source_file: /home/ge/.vlang/thirdparty/libbacktrace/backtrace.c
> wd before cmd: /tmp
> getwd for cmd: /home/ge/.vlang
>           cmd: '/home/ge/.vlang/thirdparty/tcc/tcc.exe' -std=c99 -D_DEFAULT_SOURCE   -fwrapv  -fPIC -D GC_THREADS=1 -D GC_BUILTIN_ATOMIC=1 -I"/home/ge/.vlang/thirdparty/libgc/include" -I"/home/ge/.vlang/thirdparty/libbacktrace" -o "/home/ge/.vmodules/.cache/73/73ee7692c076217496f1eb3a6d2f44ef.module.builtin.o" -c "/home/ge/.vlang/thirdparty/libbacktrace/backtrace.c"
builder error: In file included from /home/ge/.vlang/thirdparty/libbacktrace/backtrace.c:145:
In file included from /home/ge/.vlang/thirdparty/libbacktrace/linux.c:11:
In file included from /usr/include/link.h:35:
/usr/include/bits/link.h:99: error: ';' expected (got "__int128_t")
```

### Possible Solution

_No response_

### Additional Information/Context

Generated by `v bug` from `foo.v`.

### V version

V 0.5.1 3690f88

### Environment details (OS name and version, etc.)

```
|V full version      |V 0.5.1 2f3c0f1f397c08bfc5ef73e7ff62794471b2b811.3690f88
|:-------------------|:-------------------
|OS                  |linux, Linux version 7.0.11-arch1-1 (linux@archlinux) (gcc (GCC) 16.1.1 20260430, GNU ld (GNU Binutils) 2.46.0) #1 SMP PREEMPT_DYNAMIC Tue, 02 Jun 2026 18:26:58 +0000
|Processor           |16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P
|Memory              |4.82GB/15.34GB
|                    |
|V executable        |/home/ge/.vlang/v
|V last modified time|2026-06-10 13:35:52
|                    |
|V home dir          |OK, value: /home/ge/.vlang
|VMODULES            |OK, value: /home/ge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /tmp
|                    |
|Git version         |git version 2.54.0
|V git status        |weekly.2026.03-2253-g3690f882
|.git/config present |true
|                    |
|cc version          |cc (GCC) 16.1.1 20260430
|gcc version         |gcc (GCC) 16.1.1 20260430
|clang version       |clang version 22.1.6
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.43
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/builder/gc_flags_test.v
+++ b/vlib/v/builder/gc_flags_test.v
@@ -58,3 +58,71 @@ fn test_no_gc_thread_local_alloc_uses_source_libgc_without_tla_define() {
 	assert normalized.contains('-D GC_THREADS=1')
 	assert !normalized.contains('THREAD_LOCAL_ALLOC')
 }
+
+fn tcc_compiler_for_test() string {
+	bundled_tcc := os.join_path(@VEXEROOT, 'thirdparty', 'tcc', 'tcc.exe')
+	if tcc_compiler_is_usable(bundled_tcc) {
+		return bundled_tcc
+	}
+	system_tcc := os.find_abs_path_of_executable('tcc') or { return '' }
+	if tcc_compiler_is_usable(system_tcc) {
+		return system_tcc
+	}
+	return ''
+}
+
+fn tcc_compiler_is_usable(tcc_path string) bool {
+	if tcc_path == '' || !os.is_file(tcc_path) || !os.is_executable(tcc_path) {
+		return false
+	}
+	probe := os.execute('${os.quoted_path(tcc_path)} -v')
+	return probe.exit_code == 0
+}
+
+fn tcc_can_compile_v_program(tcc_path string, test_dir string) bool {
+	exe_path := os.join_path(test_dir, 'tcc_probe')
+	source_path := os.join_path(@VEXEROOT, 'examples', 'hello_world.v')
+	res :=
+		execute_without_vflags('${os.quoted_path(@VEXE)} -cc ${os.quoted_path(tcc_path)} -gc none -no-retry-compilation -o ${os.quoted_path(exe_path)} ${os.quoted_path(source_path)}')
+	return res.exit_code == 0
+}
+
+fn test_tcc_use_libbacktrace_does_not_compile_libbacktrace() {
+	test_dir := os.join_path(os.vtmp_dir(), 'builder_use_libbacktrace_tcc_${os.getpid()}')
+	os.mkdir_all(test_dir) or { panic(err) }
+	defer {
+		os.rmdir_all(test_dir) or {}
+	}
+	tcc_path := tcc_compiler_for_test()
+	vmodules_path := os.join_path(test_dir, 'vmodules')
+	old_vmodules := os.getenv_opt('VMODULES')
+	old_vcache := os.getenv_opt('VCACHE')
+	os.setenv('VMODULES', vmodules_path, true)
+	os.setenv('VCACHE', os.join_path(vmodules_path, '.cache'), true)
+	defer {
+		if vcache := old_vcache {
+			os.setenv('VCACHE', vcache, true)
+		} else {
+			os.unsetenv('VCACHE')
+		}
+		if vmodules := old_vmodules {
+			os.setenv('VMODULES', vmodules, true)
+		} else {
+			os.unsetenv('VMODULES')
+		}
+	}
+	if tcc_path == '' || !tcc_can_compile_v_program(tcc_path, test_dir) {
+		return
+	}
+	source_path := os.join_path(test_dir, 'main.v')
+	exe_path := os.join_path(test_dir, 'main')
+	os.write_file(source_path, "fn main() {\n\tpanic('aaaa')\n}\n") or { panic(err) }
+
+	res :=
+		execute_without_vflags('${os.quoted_path(@VEXE)} -cc ${os.quoted_path(tcc_path)} -gc none -no-retry-compilation -d use_libbacktrace -d trace_thirdparty_obj_files -o ${os.quoted_path(exe_path)} ${os.quoted_path(source_path)}')
+
+	assert res.exit_code == 0, res.output
+	normalized := res.output.replace('\\', '/')
+	assert !normalized.contains('thirdparty/libbacktrace/backtrace'), res.output
+	assert !normalized.contains('Failed build_thirdparty_obj_file'), res.output
+}

